src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchReaderStream.cs

```diff
@@ -0,0 +1,680 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataMultipartMixedBatchReaderStream.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.MultipartMixed
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Text;
+
+    /// <summary>
+    /// Class used by the <see cref="ODataMultipartMixedBatchReader"/> to read the various pieces of a batch payload
+    /// in multipart/mixed format.
+    /// </summary>
+    /// <remarks>
+    /// This stream separates a batch payload into multiple parts by scanning ahead and matching
+    /// a boundary string against the current payload.
+    /// </remarks>
+    internal sealed class ODataMultipartMixedBatchReaderStream : ODataBatchReaderStream

```

Comment:
>ODataBatchReaderStream [](start = 65, length = 22)

What is the benefit of having this base class? It looks like the derived class creates the stream and uses it without ever exposing it to the base class?

Reply:
This base class defines common functionality between MIME batch stream and Json batch stream, such as byte buffer and encoding decoding.
The derived class such as ODataMultipartMixedBatchReaderStream defines format specific reading functionality, such as raw MIME reader or Json reader, from its associated HTTP stream from the request and response.

---
In reply to: [145856175](https://github.com/OData/odata.net/pull/960#discussion_r145856175) [](ancestors = 145856175)